### PR TITLE
BUG Fix: Series.add(DataFrame) fails to align

### LINKED ISSUE #46179: BUG: Series.add(DataFrame) fails to align
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd

df = pd.DataFrame(
    {2010: [1], 2020: [3]},
    index=pd.MultiIndex.from_product(
        [["a"], ["b"]], names=["scen", "mod"]
    ),
)

s = pd.Series(
    [10., 20., 30.],
    index=pd.MultiIndex.from_product([["a"], ["b"], [0, 1, 2]], names=["scen", "mod", "id"]),
)

print(s.add(df, axis=0))
```

### Issue Description

The addition operation fails with

```
Traceback (most recent call last):
  File "/home/jmu3si/tmp/addtest.py", line 15, in <module>
    print(s.add(df, axis=0))
  File "/home/jmu3si/Devel/pandas/pandas/core/ops/__init__.py", line 197, in flex_wrapper
    return op(self, other)
  File "/home/jmu3si/Devel/pandas/pandas/core/ops/common.py", line 70, in new_method
    return method(self, other)
  File "/home/jmu3si/Devel/pandas/pandas/core/arraylike.py", line 104, in __radd__
    return self._arith_method(other, roperator.radd)
  File "/home/jmu3si/Devel/pandas/pandas/core/frame.py", line 7019, in _arith_method
    self, other = ops.align_method_FRAME(self, other, axis, flex=True, level=None)
  File "/home/jmu3si/Devel/pandas/pandas/core/ops/__init__.py", line 307, in align_method_FRAME
    left, right = left.align(
  File "/home/jmu3si/Devel/pandas/pandas/core/frame.py", line 4719, in align
    return super().align(
  File "/home/jmu3si/Devel/pandas/pandas/core/generic.py", line 8872, in align
    return self._align_series(
  File "/home/jmu3si/Devel/pandas/pandas/core/generic.py", line 8989, in _align_series
    join_index, lidx, ridx = join_index.join(
  File "/home/jmu3si/Devel/pandas/pandas/core/indexes/base.py", line 221, in join
    join_index, lidx, ridx = meth(self, other, how=how, level=level, sort=sort)
  File "/home/jmu3si/Devel/pandas/pandas/core/indexes/base.py", line 4541, in join
    return self._join_multi(other, how=how)
  File "/home/jmu3si/Devel/pandas/pandas/core/indexes/base.py", line 4662, in _join_multi
    raise ValueError("cannot join with no overlapping index names")
ValueError: cannot join with no overlapping index names
```

### Expected Behavior

The `s.add(df, axis=0)` should yield the same result as `df.add(s, axis=0)` which results in

```
             2010  2020
scen mod id            
a    b   0   11.0  13.0
         1   21.0  23.0
         2   31.0  33.0
```

### Installed Versions

<details>
INSTALLED VERSIONS
------------------
commit           : d801a4b19dd7f3d1ad2ef984114cb7e7389d4700
python           : 3.9.7.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.0-100-lowlatency
Version          : #113-Ubuntu SMP PREEMPT Thu Feb 3 19:24:13 UTC 2022
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : de_DE.UTF-8
LOCALE           : de_DE.UTF-8

pandas           : 1.5.0.dev0+443.gd801a4b19d
numpy            : 1.21.4
pytz             : 2021.3
dateutil         : 2.8.2
pip              : 21.2.4
setuptools       : 58.0.4
Cython           : 0.29.24
pytest           : 6.2.5
hypothesis       : 6.24.6
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : 3.0.2
lxml.etree       : 4.8.0
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 3.0.3
IPython          : 7.29.0
pandas_datareader: None
bs4              : 4.10.0
bottleneck       : None
fastparquet      : None
fsspec           : 2022.01.0
gcsfs            : None
markupsafe       : 2.0.1
matplotlib       : 3.5.1
numba            : 0.55.1
numexpr          : 2.8.1
odfpy            : None
openpyxl         : 3.0.9
pandas_gbq       : None
pyarrow          : 7.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.8.0
sqlalchemy       : 1.4.31
tables           : 3.7.0
tabulate         : None
xarray           : 0.21.1
xlrd             : 2.0.1
xlwt             : 1.3.0
zstandard        : None
</details>

### PR DESCRIPTION
**Pull request by weikhor**
- [ ] closes #46179
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by phofl — COMMENTED** [review]
Not sure, if this should work. The docstring says Series or scalar. Can you check how this is annotated?

File: doc/source/whatsnew/v1.5.0.rst [hunk 1]
```diff
@@ -855,6 +855,7 @@ Numeric
 - Bug in division, ``pow`` and ``mod`` operations on array-likes with ``dtype="boolean"`` not being like their ``np.bool_`` counterparts (:issue:`46063`)
 - Bug in multiplying a :class:`Series` with ``IntegerDtype`` or ``FloatingDtype`` by an array-like with ``timedelta64[ns]`` dtype incorrectly raising (:issue:`45622`)
 - Bug in :meth:`mean` where the optional dependency ``bottleneck`` causes precision loss linear in the length of the array. ``bottleneck`` has been disabled for :meth:`mean` improving the loss to log-linear but may result in a performance decrease. (:issue:`42878`)
+- Bug in :meth:`flex_method_SERIES` Series add with DataFrame fails to align (:issue:`46179`)
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.5.0.rst:858]: `flex_method_SERIES` is an implementation detail. please reword in terms of using the public api

**Comment by simonjayhawkins** [comment]
> Not sure, if this should work. The docstring says Series or scalar. Can you check how this is annotated?

agreed. The error message should perhaps be more informative.

Seems to be a long-standing behavior. would probably need to check for similar issues or other examples which would suggest that this should work. I'm not sure we guarantee associativity for addition of these mixed classes. Also the flex methods are also for methods that are not associative so may need also to parametrize the test over all the flex methods if doing this change. As I assume from the issue, that's the reasoning for expecting `Series.add(DataFrame)` to work?

The docstring defines our public api, so also this would perhaps be an enhancement and not a bug.

I think we should also maybe push the discussion on expected behavior back to the issue.

**Comment by mroeschke** [comment]
Agreed, let's continue the discussion to the issue before moving forward with the PR. Thanks @weikhor but feel free to contribute thoughts to the issue, closing for now.

**Comment by weikhor** [comment]
@simonjayhawkins @mroeschke @phofl  Thank :)

**Pull request closed without merging**